socketio/src/socket.rs

```diff
@@ -197,24 +193,16 @@ impl Socket {
 
         // the packet, encoded as an engine.io message packet
         let engine_packet = EnginePacket::new(EnginePacketId::Message, Bytes::from(packet));
+        self.engine_socket.emit(engine_packet)?;
 
-        self.engine_socket.emit(engine_packet, false)?;
-
-        Ok(())
-    }
-
-    /// Sends a single binary attachment to the server. This method
-    /// should only be called if a `BinaryEvent` or `BinaryAck` was
-    /// send to the server mentioning this attachment in it's
-    /// `attachments` field.
-    fn send_binary_attachment(&self, attachment: Bytes) -> Result<()> {
-        if !self.is_engineio_connected()? || !self.connected.load(Ordering::Acquire) {
-            return Err(Error::IllegalActionBeforeOpen());
+        if let Some(ref attachments) = packet.attachments {
+            for attachment in attachments {
+                let engine_packet =
+                    EnginePacket::new(EnginePacketId::MessageBinary, attachment.clone());
```

Comment:
Why do we clone here? It should be possible to partially move ownership of the bytes here, what do you think?

Reply:
Good catch, the send function should take ownership of the packet rather than just a reference. 30675a2cce96b3c6db4f1f1df0b75764c1d859e7